akinyeleolat/politico
# #163654303 Bug Fix API Bug

### What does this PR do?
Fix Bug identified

### Description of Task to be completed?
- correct the validations errors
- add missing parameters
### How should this be manually tested?
clone repo
npm install
npm test
### Any background context you want to provide?
### Expected
- Add hqAddress on party objects
- include same in your UI.
- take care of undefined in req.body
- fix validation to avoid numbers in party and office name and type
- return the parameters in empty req.body
- take care of white spaces in get specific item
- make error an object to return more than one error if possible
- validation for string should be string only
- set your errors and json response from users perspective

### Actual
- hq Address is absent
- req.body throws error in undefined
- party name is accepting number
- white spaces is allowed in req.params
- errors are returned individually
- string is accepting other number as part e.g "chairma2@"

### What are the relevant pivotal tracker stories?
163654303
![image](https://user-images.githubusercontent.com/19696366/52150163-650bac80-266f-11e9-9e64-5ff63ff375cf.png)
